cmd/statusd/main.go

```diff
@@ -248,6 +249,11 @@ func makeNodeConfig() (*params.NodeConfig, error) {
 	nodeConfig.LightEthConfig.Enabled = *lesEnabled
 	nodeConfig.SwarmConfig.Enabled = *swarmEnabled
 
+	nodeConfig.StatusServiceEnabled = *statusServiceEnabled
+	if nodeConfig.StatusServiceEnabled {
+		nodeConfig.AddAPIModule("status")
```

Comment:
Why do we need to do it? If `status` API is private and the SDK will communicate via IPC, it should not be required.

Reply:
@adambabik @adriacidre If `status` is hardcoded in the `APIModules`, this test is failing: https://github.com/status-im/status-go/blob/develop/t/e2e/services/status_api_test.go#L56

Reply:
so is it better to remove that test or better leave the kind of configuration?

Reply:
> That test is checking exactly that those endpoints are not publicly accessible and in consequence accessible from any random dApp.

"and in consequence **not** accessible from any random dApp" it should say, right?

Reply:
yeah that's I add it from `statusd`. If I leave it in the `APIModules` that test fails, if I don't add it from `statusd`, those methods are not available via HTTP/IPC

Reply:
In the past, I proposed to skip the signup/login process to avoid security problems in the SDK and leave the generation of the keys to the user or let bots use keys with `shh_generateKeyPair()` https://github.com/status-im/status-go/pull/877#issuecomment-385924009. but since we decided to support that service I made these changes

Reply:
Otherwise, we can use `nodeConfig.AddAPIModule("status")` only when `-http` is disabled, and `-ipc` is enabled. So that we ensure these API calls are possible only from local.